[webkitpy] Never use threading when running layout tests

JonWBedard opened the pull request:
#### 56c4f2bf46d0a08504f93ab0085502d800b06a70
<pre>
[webkitpy] Never use threading when running layout tests
<a href="https://bugs.webkit.org/show_bug.cgi?id=297436">https://bugs.webkit.org/show_bug.cgi?id=297436</a>
<a href="https://rdar.apple.com/158353329">rdar://158353329</a>

Reviewed by NOBODY (OOPS!).

We should never use threads in run-webkit-tests. Remove threading, replace with the Timeout class (which uses
an interupt).

* Tools/Scripts/webkitpy/layout_tests/controllers/layout_test_runner.py:
(Worker.run_test): Invoke _run_test_with_timeout instead of _run_test_with_or_without_timeout.
(Worker._run_test_with_timeout): Renamed from Worker._run_test_with_or_without_timeout. Remove threading, but use Timeout.
(Worker._run_test_with_or_without_timeout): Deleted.
(Worker._run_test_in_another_thread): Deleted.
(Worker._run_test_in_another_thread.SingleTestThread): Deleted.
(Worker._run_test_in_another_thread.SingleTestThread.__init__): Deleted.
(Worker._run_test_in_another_thread.SingleTestThread.run): Deleted.
(Worker._run_test_in_this_thread): Deleted.
</pre><!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/56c4f2bf46d0a08504f93ab0085502d800b06a70

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/118515 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/38196 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/28847 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/124687 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/70574 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/d37516f7-ad2e-4abf-8ebe-a4bd12a566f7) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/120393 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/38892 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/46778 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/89956 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/59524 "Passed tests") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/958b37be-7a65-46bb-b38c-34b331d71022) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/121468 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/30979 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/106261 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/70458 "Passed tests") | | [❌ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/a817772e-4f30-4649-ab36-79662e287529) 
| [❌ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/117875 "Found 1 webkitpy test failure: webkitpy.layout_tests.run_webkit_tests_integrationtest.RunTest.serial_test_run_singly_actually_runs_tests") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/30037 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/24373 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/68346 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/100421 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/24561 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/127753 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/45422 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/34271 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/98601 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/45786 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/102480 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/98385 "Passed tests") | | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/43811 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/21804 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/41920 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/45292 "Built successfully") | | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/44755 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/48102 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/46442 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

gsnedders reviewed (changes requested):
Nothing seems to cause a new driver to be created when `--run-singly` is passed after this change? With `--run-singly`, we presumably also want to kill the driver regardless of whether `self._driver.has_crashed()`.

gsnedders commented on Tools/Scripts/webkitpy/layout_tests/run_webkit_tests_integrationtest.py:
```diff
@@ -445,7 +445,7 @@ def serial_test_skip_failing_tests(self):
     def serial_test_run_singly_actually_runs_tests(self):
         details, _, _ = logging_run(['--run-singly'], tests_included=True)
-        self.assertEqual(details.exit_code, test.UNEXPECTED_FAILURES - 3)  # 3 tests succeed on retry
+        self.assertEqual(details.exit_code, test.UNEXPECTED_FAILURES - 4)  # 4 tests succeed on retry
```
Why does this change? I really wouldn't expect this to change as a result?

JonWBedard committed:
[webkitpy] Never use threading when running layout tests
https://bugs.webkit.org/show_bug.cgi?id=297436
rdar://158353329

Reviewed by NOBODY (OOPS!).

We should never use threads in run-webkit-tests. Remove threading, replace with the Timeout class (which uses
an interupt).

* Tools/Scripts/webkitpy/layout_tests/controllers/layout_test_runner.py:
(Worker.run_test): Invoke _run_test_with_timeout instead of _run_test_with_or_without_timeout.
(Worker._run_test_with_timeout): Renamed from Worker._run_test_with_or_without_timeout. Remove threading, but use Timeout.
(Worker._run_test_with_or_without_timeout): Deleted.
(Worker._run_test_in_another_thread): Deleted.
(Worker._run_test_in_another_thread.SingleTestThread): Deleted.
(Worker._run_test_in_another_thread.SingleTestThread.__init__): Deleted.
(Worker._run_test_in_another_thread.SingleTestThread.run): Deleted.
(Worker._run_test_in_this_thread): Deleted.

gsnedders reviewed (approved):
This looks fine, now.

Though I will point out that avoiding threads entirely is probably unrealistic, because enough different things _do_ practically start threads, whether within Python or within C code.
